### ISSUE TITLE
Add more options to customize widget font

### ISSUE THREAD

**Issue by vipaoL** [BlackyHawky/Clock#122]
I want to make the font less bold. It would be great to add the option to adjust font thickness and maybe some other options like font style and the option to choose a custom font

**Comment by moriel5** [comment]
Agreed, while personally I do not use the widget, I have seen people (such as my sister) preferring alternative widgets, just because of this.

**Comment by BlackyHawky** [comment]
Hi everyone,

I don't plan to do this for two reasons:

1. A widget is displayed on the launcher, so I believe it should follow the device's font and nothing else;

2. Technically, to change the font of the `TextClock` (derived from `TextView`), the `setTypeface()` method isn't one of the methods where `RemotableViewMethod` is specified; this means I can't apply it to our `Remoteview` that manages the widget. <br>
An alternative solution would be to create a `Bitmap` to display the time and use the `RemoteViews.setImageViewBitmap()` method.
However, this means the widget must update every minute (or even every second if the user has chosen this in the widget settings) to reflect the new time using the font.
You'll understand that this would significantly impact the device's performance and battery life ; that's why I will never do this.

Maybe a more experienced developer has another solution.

**Comment by f4646** [comment]
@BlackyHawky 
You mean that changing the font dynamically isn't possible because setTypeface() is not supported by RemoteViews.

Would it be possible to work around this by creating separate widget layouts with different predefined fonts and letting users choose between them? For example, offering a font similar to the Nothing NDot typeface used on Nothing phones.

**Comment by BlackyHawky** [comment]
> [@BlackyHawky](https://github.com/BlackyHawky) You mean that changing the font dynamically isn't possible because setTypeface() is not supported by RemoteViews.
> 
> Would it be possible to work around this by creating separate widget layouts with different predefined fonts and letting users choose between them? For example, offering a font similar to the Nothing NDot typeface used on Nothing phones.

Sorry, but I won't be making any changes related to the font used for widgets.

You might want this font, but another user might want a different one, and in the end, it would create a mess in terms of the app's code and maintenance.

Even today, the only proper solution is for Google to create/update an API that allows `setTypeface()` to be applied to widgets.
